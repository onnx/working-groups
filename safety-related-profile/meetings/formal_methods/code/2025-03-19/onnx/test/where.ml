let pp_real fmt v = Format.fprintf fmt "%3.2f" v
let pp_bool fmt v = Format.fprintf fmt "%c" (if v then 'T' else 'F')

let c = Tensor.vector [true;true;false;true;false]
let a = Tensor.vector [1.0;2.0;3.0;4.0;5.0]
let b = Tensor.vector [0.1;0.2;0.3;0.4;0.5]
let w = Tensor.where c a b

let () =
  begin
    Format.printf "C: %a@." (Tensor.pretty pp_bool) c ;
    Format.printf "A: %a@." (Tensor.pretty pp_real) a ;
    Format.printf "B: %a@." (Tensor.pretty pp_real) b ;
    Format.printf "W: %a@." (Tensor.pretty pp_real) w ;
   end
