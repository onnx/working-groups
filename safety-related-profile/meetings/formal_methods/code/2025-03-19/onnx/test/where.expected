C: [ T T F T F ]
A: [ 1.00 2.00 3.00 4.00 5.00 ]
B: [ 0.10 0.20 0.30 0.40 0.50 ]
W: [ 1.00 2.00 0.30 4.00 0.50 ]
